src/Struct/FindologicProduct.php

```diff
@@ -762,14 +762,30 @@ protected function setOrdernumberByProduct(ProductEntity $product): void
     protected function setCategoriesAndCatUrls(): void
     {
         $productCategories = $this->product->getCategories();
-        if ($productCategories === null || empty($productCategories->count())) {
+        $productVariants = $this->product->getChildren();
+
+        // If there are no product categories, and no children either, we throw ProductHasNoCategoriesException.
```

Comment:
This comment is rather useless, reading the line below is already clear enough. If it wasn't, or if you aren't sure someone else would get what it does, maybe refactor this into a small method with a clearer name. For me personally the code is already self-explanatory.